core/corehttp/gateway.go

```diff
@@ -89,9 +90,10 @@ func GatewayOption(writable bool, paths ...string) ServeOption {
 			}, headers[ACEHeadersName]...))
 
 		var gateway http.Handler = newGatewayHandler(GatewayConfig{
-			Headers:      headers,
-			Writable:     writable,
-			PathPrefixes: cfg.Gateway.PathPrefixes,
+			Headers:             headers,
+			Writable:            writable,
+			PathPrefixes:        cfg.Gateway.PathPrefixes,
+			HTMLDirListingLimit: int(cfg.Gateway.HTMLDirListingLimit.WithDefault(2500)),
```

Comment:
@alanshaw is 2500 a good default limit, or should we increase it?

Reply:
Just to make sure this is the expected UX: if the user has no config set on this, it will silently use this hard-coded number. (Not opposed, just checking.)